### PR TITLE
Driver timestamp

### PR DESCRIPTION
**Pull request by guiangumpac**
### Description
- Added type timestamp to SQLGetTypeInfo
- Modified a test to include changes
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/317
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sql-odbc/src/sqlodbc/opensearch_info.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ const std::unordered_map< std::string, int > data_name_data_type_map = {
     {OPENSEARCH_TYPE_NAME_SCALED_FLOAT, SQL_DOUBLE},
     {OPENSEARCH_TYPE_NAME_KEYWORD, SQL_WVARCHAR},
     {OPENSEARCH_TYPE_NAME_TEXT, SQL_WVARCHAR},
-    {OPENSEARCH_TYPE_NAME_DATE, SQL_TYPE_TIMESTAMP},
+    {OPENSEARCH_TYPE_NAME_TIMESTAMP, SQL_TYPE_TIMESTAMP},
```
Comment (kylepbit) [inline sql-odbc/src/sqlodbc/opensearch_info.cpp:89]: Shouldn't you keep the date mapping to timestamp here as well?

File: sql-odbc/src/sqlodbc/opensearch_types.c [hunk 2]
```diff
@@ -372,7 +372,7 @@ opensearchtype_attr_to_sqldesctype(const ConnectionClass *conn, OID type, int ty
         case SQL_TYPE_DATE:
         case SQL_TYPE_TIME:
         case SQL_TYPE_TIMESTAMP:
-            return SQL_DATETIME;
+            return SQL_TYPE_TIMESTAMP;
```
Comment (kylepbit) [inline sql-odbc/src/sqlodbc/opensearch_types.c:375]: I am not sure this is correct, I believe SQL_DATETIME is the correct return here?

File: sql-odbc/src/sqlodbc/opensearch_info.cpp (hunk 1 above)
Comment (guiangumpac) [reply sql-odbc/src/sqlodbc/opensearch_info.cpp:90]: Yes, I've reverted it back.

File: sql-odbc/src/sqlodbc/opensearch_types.c (hunk 2 above)
Comment (guiangumpac) [reply sql-odbc/src/sqlodbc/opensearch_types.c]: This has be reverted in [1817773](https://github.com/Bit-Quill/opensearch-project-sql/pull/15/commits/1817773228666e95818487f06aecdaba7caab8eb)

**Review by kylepbit — APPROVED** [review]

**Pull request merged**